ElevatorSim/src/elevatorsim/scheduler/SchedulerServer.java

```diff
@@ -45,37 +54,102 @@ public Role getRole() {
 	/**
 	 * Forwards An Elevator Request to a Registered Elevator that can Service it
 	 */
+	
+	public boolean handleElevatorCall(ElevatorRequest elevatorRequest) {
+		 // + elevatorRequest. + "an elevator is requested at floor" + elevatorRequest.getStartFloor().toString() + ". (The passenger will travel " + elevatorRequest.getDirection().toString() + "to floor " + Integer.toString(elevatorRequest.getDestFloor()).toString() + ".");
+		InetAddress availableElevator = Scheduler.getInstance().findAvailableElevator(elevatorRequest.getStartFloor(), elevatorRequest.getDirection());
+		
+		//If an elevator isnt available, store the request
+		//If there is an available elevator but it's moving then ignore it too for now because it's not worth optimising until there are multiple elevators
+		if (availableElevator == null || Scheduler.getInstance().getElevators().get(availableElevator).getDirection() != null ) {
+			ArrayList<ElevatorRequest> storedRequests = Scheduler.getInstance().getStoredRequests();
+			storedRequests.add(elevatorRequest);
+			InetAddress elevatorAddress = Scheduler.getInstance().findAvailableElevator();
+			ElevatorStatus elevatorStatus = Scheduler.getInstance().getElevators().get(elevatorAddress);
+			if( storedRequests.size() == 1 && elevatorStatus.getDirection() == null ) {
+				elevatorStatus.addFloor(elevatorRequest.getStartFloor());
+				
+				elevatorStatus.setDirection( elevatorRequest.getStartFloor() - elevatorStatus.getFloor() > 0 ? Direction.UP : Direction.DOWN );
+				sender.send(MessagePackets.generateElevatorStateChange(new ElevatorStateChange(ElevatorState.DOOR_CLOSED)), elevatorAddress, elevators.get(elevatorAddress));
+			}
+			return true;
+		}
+		System.out.println();
+		
+		
+		int elevatorStartFloor = Scheduler.getInstance().getElevators().get(availableElevator).getFloor();
+		int boardFloor = elevatorRequest.getStartFloor();
+		int floorChange = boardFloor - elevatorStartFloor;
+		Direction direction;
+		if( floorChange < 0 ) {
+			direction = Direction.DOWN;
+		} else if (floorChange > 0) {
+			direction = Direction.UP;
+		}else {
+			direction = null;
+		}
+		if (direction == null) {
+			//send request to elevator so it can send back it's version
+			Integer elevatorPort = elevators.get(availableElevator);
+			sender.send(MessagePackets.generateElevatorRequest(elevatorRequest), availableElevator, elevatorPort);
+		}
+	
+		return true;
```

Comment:
all you seem to do in this method is return true. I don't think we need a return in that case

Reply:
changed return type to void